### ISSUE TITLE
Hibernate statistics are not exposed even if corresponding configuration are applied

### ISSUE THREAD

**Issue by klopfdreh** [spring-cloud/spring-cloud-dataflow#4791]
**Description:**
As mentioned in the title, the hibernate statistics are not exposed even if I set the property `spring.jpa.properties.hibernate.generate_statistics=true`  in the application.yaml / application.properties

**Release versions:**
2.9.1

**Steps to reproduce:**
1. Just start the server and set the following properties in the application.yml / application.properties.
yaml-example:
```yaml
    management:
      endpoints:
        prometheus:
          enabled: true
        web:
          exposure:
            include: "*"
        jmx:
          exposure:
            include: "*"
      metrics:
        tags:
          application: scdf_server
        export:
          prometheus:
            enabled: true
            rsocket:
              enabled: true
              host: localhost
              port: 7001
    spring:
        jpa:
          properties:
            hibernate:
              dialect: <yourdialect>
              generate_statistics: true
```
2. Access http://localhost:9393/management/prometheus and search for example for `hibernate_query_executions_max_seconds`.
3. According to hibernate this should show the query executions max seconds, but this and other properties are not exposed. 

Note: I double checked if "spring" is applied on the same level as server and management.
Note2: This was working till version 2.9.0

**Comment by klopfdreh** [comment]
Because the statistics generation of hibernate has been moved to the hibernate project `org.hibernate:hibernate-micrometer` I guess this dependency has to be added to the classpath. See: https://docs.spring.io/spring-boot/docs/2.7.8/reference/htmlsingle/#actuator.metrics.supported.hibernate

I am going to try this out and report back if this is working.

**Comment by klopfdreh** [comment]
Good news - that was the issue here - see the result:

![image](https://user-images.githubusercontent.com/980773/223670313-39f2fee9-c386-4640-9730-1be9a6952409.png)

Just add the following dependency to the pom.xml (https://github.com/spring-cloud/spring-cloud-dataflow/blob/main/spring-cloud-dataflow-server/pom.xml) 
```xml
<dependency>
    <groupId>org.hibernate</groupId>
    <artifactId>hibernate-micrometer</artifactId>
</dependency>
```

**Comment by onobc** [comment]
Awesome find @klopfdreh ! Thanks for this. We will also make sure we have test coverage in this area of the metrics so that we find it before it goes out.

**Comment by onobc** [comment]
@klopfdreh previously were the stats automatically exposed or did you have to set the `spring.jpa.properties.hibernate.generate_statistics=true` to get them to show up?

**Comment by klopfdreh** [comment]
@onobc it’s a long time ago when I configured our environment, but I set the property directly from beginning because I thought this needs to be configured. I didn’t test it without set this property. But what I noticed is that even if I set the property it stopped working to expose the statistics which was the reason for me to investigate this issue.

I would suggest to not expose the statistics by default, because it has some impacts to the server runtime.
